Geocoding readme improvements (#30)
Esri/arcgis-maps-sdk-unreal-engine-samples: sample_project/Content/SampleViewer/Samples/Geocoding/readme.md

# Find places or search for an address

> Find the place for an address or find the address of a selected location using the Geocoding service.

Reviewer: ```suggestion
Find the location for an address or find the address of a selected location using the Geocoding service.
```

![Geocoding](Geocoding.jpg)

## How to use the sample

This sample demonstrates the capability of the [Geocoding service](https://developers.arcgis.com/documentation/mapping-apis-and-services/search/geocoding/) inside Unreal Engine. You can make two kinds of operations with this sample, find a place and find an address. 

1. Open the **Geocoding** level.

> 2. Click on the **ArcGISMap Actpr** in the **Outliner**.

Reviewer: ```suggestion
2. Click on the **ArcGISMap Actor** in the **Outliner**.
```

3. Set your API key in the **Details** panel.
4. Click play.
5. To find a place, put a place name or address in the search bar and press Enter. For information about the type of locations that are supported by the Geocoding service, see the [ArcGIS REST APIs documentation](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm).
6. To find an address of a place, drop the marker to your point of interest by clicking in the viewport while holding Shift, and you will see the address displayed above the marker.

> **Note:** This sample is only set up to work with a mouse and keyboard.

Reviewer: ```suggestion
**Note:** This sample is only set up to work with mouse and keyboard.
```

## How it works

1. Have a map with the mesh colliders enabled in the level. The mesh collider is required for using raycast. 
2. Have an Actor with the C++ script to query [Geocoding service](https://developers.arcgis.com/rest/geocode/api-reference/overview-world-geocoding-service.htm).
3. The HTTPClient is used to query the Geocoding service.

> 4. The [`findAddressCandidates`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm) operation is used for finding a place.

Reviewer: ```suggestion
4. The [`findAddressCandidates`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm) API is used for finding a place.
```

Author: This is how it's expressed in the REST API doc so I think it's better to align.
https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm

![image](https://user-images.githubusercontent.com/53481408/201326802-37722e75-c9c7-4eac-8427-acf5009a1d8e.png)

> 5. The [`reverseGeocode`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-reverse-geocode.htm) operation is used for finding an address. The marker should have the [**ArcGIS Location Component**](https://developers.arcgis.com/unity/maps/location-component/) attached and its values will be used to query the address of the location.

Reviewer: ```suggestion
5. The [`reverseGeocode`](https://developers.arcgis.com/rest/geocode/api-reference/geocoding-reverse-geocode.htm) API is used for finding an address. The marker should have the [**ArcGIS Location Component**](https://developers.arcgis.com/unity/maps/location-component/) attached and its values will be used to query the address of the location.
```

> 6. Use the game engine's [raycast](https://docs.unrealengine.com/5.0/en-US/API/Runtime/Engine/Engine/UWorld/LineTraceSingleByChannel/) feature to determine the height at the queried location to account for elevation to place the marker.

Reviewer 2: ```suggestion
6. Place a marker using the game engine's [raycast](https://docs.unity3d.com/ScriptReference/Physics.Raycast.html) feature to account for elevation at the queried position.
```

## Tags

> geocode, reverse geocode, address, locate, search

Reviewer: ```suggestion
geocoding, reverse geocoding, address, locate, search
```

Reviewer: geocode is the code, geocoding is the process
